docs/how-to-write-a-contract/scriptcontext.md

```diff
@@ -28,6 +28,8 @@ export interface ScriptContext {
   locktime: bigint,
   /** [SIGHASH flag]{@link https://wiki.bitcoinsv.io/index.php/SIGHASH_flags} used by this input */
   sigHashType: SigHashType,
+  /** get the whole serialized sighash preimage */
+  serialize(): SigHashPreimage,
```

Comment:
`SigHashPreimage` should just be `ByteString`, since serialization returns just raw bytes.

Reply:
Ok, but then we also have to cast to `SigHashPreimage` while using it, since the functions in `SigHash` take in a `SigHashPreimage`  type:

```ts
const txPreimage = this.ctx.serialize() as SigHashPreimage
assert(
  SigHash.hashOutputs(txPreimage) == hash256(this.buildStateOutput(SigHash.value(txPreimage))),
  'hashOutputs check failed'
)
```

Reply:
Or we could change those functions parameter types as well...

Reply:
Nah, let's keep SigHashPreimage then to void type casting.